### PR TITLE
Update Python dependencies

### PR DESCRIPTION
**Pull request by jdavcs**
## What did you do? 
Pinned pygithub in pyptoject.toml; update all python dependencies

## Why did you make this change?
This resolves the issue discussed here https://github.com/galaxyproject/galaxy/pull/11695#pullrequestreview-616895910
This replaces #11685.

## How to test the changes? 
This is a refactoring of components with existing test coverage.

### CODE REVIEW

**Review by nsoranzo — APPROVED** [review]

**Comment by nsoranzo** [comment]
Looks like there is an issue with the lagom update.

**Comment by ic4f** [comment]
> Looks like there is an issue with the lagom update.

Oh yes, infinite recursion. I'm going to pin it and see what happens.

**Comment by ic4f** [comment]
While I was fixing it, lagom came up with 1.3.0. If this works, I'll change the pin.

**Comment by ic4f** [comment]
@nsoranzo this looks good to me; there were some intermittent failures that all passed on subsequent reruns. I've pinned lagom to <1.2.1, which still results in an upgrade (1.1.0 -> 1.2.0). The underlying problem will need to be looked into eventually, but I'd much rather leave it as is for now, so it's not a blocker for further updates/additions/etc. 
Not merging yet, as your approval was before the pin.

**Review by nsoranzo — APPROVED** [review]

**Comment by nsoranzo** [comment]
The CI errors in test/integration/test_containerized_jobs.py are due to quay.io being down yesterday.

@ic4f Can you report the infinite recursion issue upstream? I haven't found one on https://github.com/meadsteve/lagom and it would help us track when we can remove the pin.

**Comment by ic4f** [comment]
> The CI errors in test/integration/test_containerized_jobs.py are due to quay.io being down yesterday.

Thank you!
 
> @ic4f Can you report the infinite recursion issue upstream? I haven't found one on https://github.com/meadsteve/lagom and it would help us track when we can remove the pin.

Will do - I didn't find one either. I'll add a comment to this PR with a reference to the issue.

File: lib/galaxy/dependencies/dev-requirements.txt [hunk 1]
```diff
@@ -90,7 +90,7 @@ jsonpointer==2.1; python_version >= "3.6" and python_full_version < "3.0.0" or p
 jsonschema==3.2.0
 keystoneauth1==4.3.1; python_version >= "3.6"
 kombu==5.0.2; python_version >= "3.6"
-lagom==1.2.1; python_version >= "3.6" and python_version < "4.0"
+lagom==1.3.0; python_version >= "3.6" and python_version < "4.0"
```
Comment (mvdbeek) [inline lib/galaxy/dependencies/dev-requirements.txt]: Was there a error here ? The test failures looked unrelated ?

Comment (jdavcs) [reply lib/galaxy/dependencies/dev-requirements.txt]: Here's the issue: 
https://github.com/galaxyproject/galaxy/runs/2170061584#step:6:3603

**Comment by meadsteve** [comment]
Not sure if this is the right place but there's a bugfix release 1.3.1 that should resolve this

**Comment by mvdbeek** [comment]
Thanks for the update and the fix @meadsteve!

**Pull request merged**